### DISCUSSION TITLE
Error 406 while uploading attachments

### DISCUSSION THREAD

**Discussion by josipkedves** [mozts2005/ZendeskApi_v2#574] [Q&A] [answered]
Hi, since 2023-03-01 error 406 appears when uploading attachments. Version used is 3.0.11.

`Exception: The remote server returned an error: (406) Not Acceptable.Error content:  
 Resource String: uploads.json?filename=Example.jpg  + 
 File Name: Example.jpg 
 File Length: 49965
Transfer-Encoding: chunked
Connection: keep-alive
x-request-id: 7c4a2fd2ccd341a0-DUB,7c4a2fd2ccd341a0-DUB
x-runtime: 0.038039
X-Zendesk-Zorg: yes
CF-Cache-Status: DYNAMIC
Report-To: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v3?s=PlG5kLE7CoYknFXGZhJbWGoYlcO9XLoFrWebnRUyEqZqkuVKdVzGVHgnCEXYk11znlesNXreOFGrHMi5dmtNQS7jHZ8oJY3MlvTIQ1MwKp77vsSu%2BNs4Vtuf0QIuGx4pN4wE%2F4sffQ%3D%3D"}],"group":"cf-nel","max_age":604800}
NEL: {"success_fraction":0.01,"report_to":"cf-nel","max_age":604800}
CF-RAY: 7c4a2fd2ccd341a0-LHR
Content-Type: text/html; charset=UTF-8
Date: Tue, 09 May 2023 13:11:10 GMT
Set-Cookie: __cfruid=42ee1b1c872aef578bdc24483e4252ae1df0850d-1683637870; path=/; domain=.something.zendesk.com; HttpOnly; Secure; SameSite=None
Server: cloudflare

, stacktrace:    at ZendeskApi_v2.Core.RunRequest(String resource, String requestMethod, Object body, Nullable`1 timeout, Dictionary`2 formParameters)
   at ZendeskApi_v2.Requests.Attachments.UploadAttachments(IEnumerable`1 files, Nullable`1 timeout)`

**Comment by avin-madhu** [comment]
I think you should try these steps:

- Check if there are any updates or changes to the Zendesk API documentation that may affect your code. Ensure that you're using the correct API endpoint and that your request parameters are correct.

- Verify that your authentication credentials are correct and that you have the necessary permissions to upload attachments.

- Try uploading a smaller file to see if the error persists. The error message indicates that you are trying to upload a file named "Example.jpg" with a length of 49965, which could be causing issues.

- Check the headers and parameters being sent with the request to make sure they are correct and complete.

**Reply by josipkedves** [reply]
Hi, I have checked with version 3.0.12 trying to upload simple .txt file.
Error is the same 
![image](https://github.com/Speedygeek/ZendeskApi_v2/assets/11602852/4ca892b0-3c07-42b7-aa56-9b9e5f29b4c9)

I am using IAttachments.UploadAttachments(IEnumerable<ZenFile> files, int? timeout = null); method.

**Reply by mozts2005** [reply, marked as answer]
Please, create a test case that replicates the issue and send a pull request. 

here is the current test case that is passing. 

https://github.com/Speedygeek/ZendeskApi_v2/blob/94082b6d7a81bbbf127c7a443fd783b6e9f1e764/tests/ZendeskApi_v2.Tests/AttachmentTests.cs#LL17C1-L26C1

```csharp 
[Test]
public void CanUploadAttachments()
{
    var res = Api.Attachments.UploadAttachment(new ZenFile()
      {
        ContentType = = "text/plain",
        FileName = "testupload.txt",
        FileData = File.ReadAllBytes(Path.Combine(TestContext.CurrentContext.TestDirectory,"testupload.txt"))
     });
    Assert.That(!string.IsNullOrEmpty(res.Token), Is.True);
}
```

Note: The Content-Type must contain a recognized MIME type that correctly describes the type of the uploaded file.

**Comment by fuhye** [comment]
Hi

Additionally ensure your file is within the size limits for your plan.
Also, in your request I see text/html in your content type, is that correct?
Run the unit test what elizabeth gave you

test with images compare with fiddler the request sended

**Comment by josipkedves** [comment]
Hi, thank you for your answer.
The issues was that Content Type has been set to jpg, txt, png ... instead of image/jpg, text/plain, image/png ...

However, prior to 2023-03-01 attachments with these content types have been successful.
